README.md

```diff
@@ -42,7 +43,7 @@ The basic I/O primitive of SwiftNIO is the event loop. The event loop is an obje
 
 Event loops are gathered together into event loop *groups*. These groups provide a mechanism to distribute work around the event loops. For example, when listening for inbound connections the listening socket will be registered on one event loop. However, we don't want all connections that are accepted on that listening socket to be registered with the same event loop, as that would potentially overload one event loop while leaving the others empty. For that reason, the event loop group provides the ability to spread load across multiple event loops.
 
-In SwiftNIO today there is one `EventLoopGroup` implementation, and two `EventLoop` implementations. For production applications there is the `MultiThreadedEventLoopGroup`, an `EventLoopGroup` that creates a number of threads (using the POSIX `pthreads` library) and places one `SelectableEventLoop` on each one. The `SelectableEventLoop` is an event loop that uses a selector (either `kqueue` or `epoll` depending on the target system) to manage I/O events from file descriptors and to dispatch work. Additionally, there is the `EmbeddedEventLoop`, which is a dummy event loop that is used primarily for testing purposes.
+In SwiftNIO today there is one `EventLoopGroup` implementation, and two `EventLoop` implementations. For production applications there is the [`MultiThreadedEventLoopGroup`](https://apple.github.io/swift-nio/docs/current/NIO/Classes/MultiThreadedEventLoopGroup.html), an `EventLoopGroup` that creates a number of threads (using the POSIX [`pthreads`](https://en.wikipedia.org/wiki/POSIX_Threads) library) and places one `SelectableEventLoop` on each one. The `SelectableEventLoop` is an event loop that uses a selector (either [`kqueue`](https://en.wikipedia.org/wiki/Kqueue) or [`epoll`](https://en.wikipedia.org/wiki/Epoll) depending on the target system) to manage I/O events from file descriptors and to dispatch work. Additionally, there is the [`EmbeddedEventLoop`](https://apple.github.io/swift-nio/docs/current/NIO/Classes/EmbeddedEventLoop.html), which is a dummy event loop that is used primarily for testing purposes.
```

Comment:
instead of repeating the links all the time I think we should use the following markdown feature:

```markdown

[`EventLoopGroup`][elg] ... [`EventLoopGroup`][elg]

...
...
...


[elg]: https://apple.github.io/swift-nio/docs/current/NIO/Classes/MultiThreadedEventLoopGroup.html
```

which then renders like this (so the links all point to the same URL `elg` is basically an abbreviation for the URL then)


[`EventLoopGroup`][elg] ... [`EventLoopGroup`][elg]

...
...
...


[elg]: https://apple.github.io/swift-nio/docs/current/NIO/Classes/MultiThreadedEventLoopGroup.html

Reply:
Not applicable to `MultiThreadedEventLoopGroup` as it's mentioned only once, but will apply this to other types. Thanks for the tip 👍

Reply:
@maxdesiatov tbh I'd do it for all important things. We can collect a list of all the more important things at the very bottom and then reference them. Looks better in the source code I think.

Reply:
@weissi would you still like links to be present at every mention of a type in `README.md` or only on the first use?

Reply:
@maxdesiatov sorry, I don't quite understand the question. You can just define shorthands and I suggest the very bottom of the `README.md` page to put all the URL shorthands.

So basically

```
Bla Bla some text [`EventLoopGroup`][elg] bla bla [`EventLoopGroup`][elg]
more text

bla bla


[elg]: https://link/to/ELG
[mtelg]: https://link/to/MultiThreadedELG
...
```

So for stuff that might be mentioned more than once I'd generally replace `[...](https://....)` by `[...][shorthand]` and add `[shorthand]: https://...` to the bottom of the page. But obviously don't feel obliged to do all that in your PR, every little bit helps :)